core/src/main/java/com/hcl/ecomm/core/servlets/GetWishListItemsServlet.java

```diff
@@ -37,25 +40,28 @@ protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse r
 
          try {
             String customerToken = request.getHeader("CustomerToken");
-            JsonObject responseStream = wishListService.getWishListItems(customerToken);
+            JSONObject responseStream = getWishListItems(customerToken);
              LOG.debug("responseStream is {}", responseStream.toString());
-        
-            JsonArray wishlistitems= responseStream.get("items").getAsJsonArray();
+             JSONArray wishlistitems= responseStream.getJSONArray("items");
             List<HashMap<String, Object>> list = new ArrayList<>();
-            JsonArray wishlistArray=new JsonArray();
+            JSONArray wishlistArray=null;
 
-            for(int i=0; i<wishlistitems.size();i++){
+            for(int i=0; i<wishlistitems.length();i++){
                 HashMap<String, Object> itemdetails = new HashMap<String, Object>();
-                itemdetails.put("item_id",wishlistitems.get(i).getAsJsonObject().get("id").getAsInt());
-                itemdetails.put("sku", wishlistitems.get(i).getAsJsonObject().get("product").getAsJsonObject().get("sku").getAsString());
-                itemdetails.put("name", wishlistitems.get(i).getAsJsonObject().get("product").getAsJsonObject().get("name").getAsString());
-                itemdetails.put("price",wishlistitems.get(i).getAsJsonObject().get("product").getAsJsonObject().get("price").getAsInt());
+                JSONObject Obj = wishlistitems.getJSONObject(i);
+                LOG.info("wishlistitems is {}", Obj);
+                itemdetails.put("item_id",Obj.get("item_id"));
+                itemdetails.put("sku", Obj.get("sku"));
+                itemdetails.put("name", Obj.get("name"));
+                itemdetails.put("price",Obj.get("price"));
```

Comment:
Will this servlet not break in case of null values? @uditnaagar777 
Please make the changes to handle the null checks.

Reply:
Thanks Shubham, The null check was tested for this code, Servlet will not break here.